device/core/test/_module_client_test.js

```diff
@@ -49,24 +50,39 @@ describe('ModuleClient', function () {
   });
 
   describe('#fromEnvironment', function () {
-    // Tests_SRS_NODE_MODULE_CLIENT_13_033: [ The fromEnvironment method shall throw a ReferenceError if the callback argument is not a function. ]
+    [{}, 20].forEach(function (badArgument) {
+      it('throws if the second argument is neither a string nor a function', function () {
```

Comment:
This makes me think that we should 100% stop overloading functions as soon as we can (like v.next). Do you agree?

Reply:
Because I can't read these tests (or the implementations themselves) and know that they're correct without pulling out pencil and paper and giving myself a headache by going through the entire set of possibilities.  If we decide that we do want to overload functions, we should find a better way to do it.

Reply:
Hmm, I'm not sure if I agree with 100% stopping overloading, but I do agree that it is very problematic here. Here, I think making the callback the last argument no matter what makes the overloading way more painful than it needs to be.

Much of our usage of overloading has to do with the presence or absence of the callback parameter. If and when we get rid of the callback APIs (say, for v next), we would do far less overloading and if we ever do have overloading for something like adding an additional optional parameter it will be less problematic.